jupyterhub/spawners.py

```diff
@@ -350,6 +350,19 @@ def get_pod_manifest(self):
             )
             self.init_containers.append(init_container)
 
+            self.lifecycle_hooks={
+                "postStart": {
```

Comment:
I wonder if the functionality of the `postStart` could not be moved to the init container.

Reply:
Unfortunately, it cannot. I tried that.

Reply:
well you could run the script inside the init container, just not as a hook. However, one thing we discussed to extend the functionality here is to use the same script to notify the user in their shell about what has happened or if something went wrong to tell them where to find the saved session.

Reply:
@jachro what's your take on this? conceptually is the autosave check different from the repo clone that happens in the init container somehow and should therefore be kept inside the notebook container? 

This actually speaks against it (from [the docs](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#container-hooks)): 

"This hook executes immediately after a container is created. However, there is no guarantee that the hook will execute before the container ENTRYPOINT"

So if the script takes some time to run, the user could do something that will cause a conflict

Reply:
This is a fair point. To be on the save side it seems reasonable to put that into `init-container`.